src/components/o-modal/index.js

```diff
@@ -0,0 +1,75 @@
+import PropTypes from '../../js/prop-types';
+import BaseComponentGlobal from '../../js/abstract/base-component-global';
+import defineOnce from '../../js/define-once';
+import styles from './index.scss';
+import template from './_template';
+import on from '../../js/on';
+import { AXA_EVENTS, EVENTS } from '../../js/ui-events';
+
+class AXAModal extends BaseComponentGlobal {
+  static tagName = 'axa-modal';
+  static propTypes = {
+    open: PropTypes.bool,
+  };
+
+  static get observedAttributes() {
+    return ['open'];
+  }
+
+  set open(value) {
+    this.setAttribute('open', value);
+  }
+
+  get open() {
+    return this.getAttribute('open');
+  }
+
+  closeModal() {
+    this.removeAttribute('open');
+  }
+
+  closeModalByClickingOutside(event) {
+    if (event.target === this.querySelector('.js-modal')) {
+      this.closeModal();
+    }
+  }
+
+  closeModalOnEsc(event) {
+    const key = event.key || event.keyCode;
+    if (key === 'Escape' || key === 'Esc' || key === 27) {
+      this.closeModal();
+    }
+  }
+
+  init() {
+    super.init({ styles, template });
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.closeModalOnEscFunction = event => this.closeModalOnEsc(event);
+    const closeModalFunction = () => this.closeModal();
+
+    const axaButtonsRaw = this.querySelectorAll('axa-button[data-modal-close]');
+    const axaButtons = Array.from(axaButtonsRaw);
+    if (axaButtons && axaButtons.length) {
+      axaButtons.forEach(button => button.addEventListener('click', closeModalFunction));
+    }
+
+    this.onEscReleased = on(document, EVENTS.KEYUP, this.closeModalOnEscFunction);
+
+    this.addEventListener('click', event => this.closeModalByClickingOutside(event));
+
+    const closeButton = this.querySelector('.js-modal-close-button');
+    closeButton.addEventListener('click', closeModalFunction);
```

Comment:
Please also use `on` for above event handlers and clean the up within `disconnectedCallback`

Reply:
@AndyOGo As far as I can tell, this is an overkill, no? Because if the element is removed, the garbage collector will clean up remaining listeners that point to that element and its children.

Reply:
Well, let's find it out. Study the specs of [`disconnectedCallback`](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-reactions:becomes-disconnected) and all related [`[CEReactions]`](https://html.spec.whatwg.org/multipage/custom-elements.html#cereactions) IDL extended attributes.

Reply:
I also think its a bit of an overkill.. never than less, after reading this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Memory_Management#Release_when_the_memory_is_not_needed_anymore i might change my mind and might be better to clean up. Sorry @raphaellueckl

Reply:
@AndyOGo ok seems that the algorithm is not clear but it should be able to handle references. As here is a recference to a dom node, i would be surprised if the garbage collector cannot handle this cleaning. Therefore for me is ok to not add additional code and relay on the GC. Unless there is a code proof that shows this, i am neutral to this so for me is ok to merge.

Reply:
I'm sure you will find it out.
I can only suggest to study the specs and observe the implementation of GC's in various JS engines like V8, SpiderMonkey, JScript, and so on

Reply:
https://v8.dev/blog/tracing-js-dom This is a good read